### PR TITLE
MRD-2323 - exception flow for booking on process

### PR DESCRIPTION
**Pull request by robinandrewdevilliers**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/client/PpudAutomationApiClient.kt [hunk 1]
```diff
@@ -42,6 +42,9 @@ class PpudAutomationApiClient(
   fun search(
     request: PpudSearchRequest,
   ): Mono<PpudSearchResponse> {
+    println("===========================================================================")
```
Comment (DarrenW-HMPPS) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/client/PpudAutomationApiClient.kt:45]: Is this needed?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/BookingMomento.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.recommendation
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class BookingMomento(
```
Comment (DarrenW-HMPPS) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/domain/makerecalldecisions/recommendation/BookingMomento.kt:6]: Typo in the name. The correct spelling is Memento

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/MrdTestDataBuilder.kt [hunk 3]
```diff
@@ -178,15 +177,10 @@ class MrdTestDataBuilder {
           ),
           bookRecallToPpud = BookRecallToPpud(
             decisionDateTime = LocalDateTime.now(),
-            isInCustody = true,
```
Comment (DarrenW-HMPPS) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/MrdTestDataBuilder.kt:181]: Why are we removing these properties?

**Review by DarrenW-HMPPS — CHANGES_REQUESTED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/client/PpudAutomationApiClient.kt (hunk 1 above)
Comment (robinandrewdevilliers) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/client/PpudAutomationApiClient.kt:45]: ? tks - surprised the linting didn't pick that up.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/MrdTestDataBuilder.kt (hunk 3 above)
Comment (robinandrewdevilliers) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/MrdTestDataBuilder.kt:181]: They are not used.

**Pull request merged**